scripts/wasm-strip.js

```diff
@@ -0,0 +1,363 @@
+        checkedMagicAndVersion = true;
+        writeStream.write(Buffer.from(buff.slice(0, 8)));
+        currOffset += 8;
+      }
+
+      while (currOffset < maxOffsetInChunk) {
+        if (nextSectionOffset === currOffset) {
+          const relativeOffset = currOffset - chunkBaseOffset;
+          if (buff.byteLength - relativeOffset < 6 /* id + max LEB128 u32 size */) {
+            prevBuffered = buff.slice(relativeOffset);
+            return;
+          }
+          const dataView = new DataView(buff);
+          const sectionType = dataView.getUint8(relativeOffset);
+          const [size, sizeOfSize] = readULeb128(dataView,
+                                                 relativeOffset + 1);
+
+          const sectionInfo = { sectionType,
+                                offset: currOffset,
+                                contentSize: size,
+                                length: size + 1 + sizeOfSize };
+          currOffset += 1 + sizeOfSize; // Section id + size
+
+          if (sectionType === CUSTOM_SECTION_ID) {
+            skipCurrentSection = true;
+            result.removedSections.push(sectionInfo);
+          } else {
+            skipCurrentSection = false;
+            writeStream.write(Buffer.from(buff.slice(relativeOffset,
+                                                     currOffset - chunkBaseOffset)));
+            result.keptSections.push(sectionInfo);
+          }
+          nextSectionOffset = size + currOffset;
+        }
+        const newOffset = Math.min(maxOffsetInChunk, nextSectionOffset);
+        if (!skipCurrentSection) {
+          writeStream.write(Buffer.from(buff.slice(currOffset - chunkBaseOffset,
+                                                   newOffset - chunkBaseOffset)));
+        }
+        currOffset = newOffset;
+      }
+    }
+  });
+}
+
+function checkMagicNumberAndVersion(buff) {
+  if (buff.byteLength < 8) {
+    throw new Error("Error: Not a valid WebAssembly file: file too short");
+  }
+  const dataView = new DataView(buff);
+  if (dataView.getUint32(0) !== WASM_MAGIC_NUMBER) {
+    throw new Error("Error: Not a valid WebAssembly file: no magic number");
+  }
+  if (dataView.getUint32(4, true) !== 1) {
+    throw new Error("Error: Unsupported WebAssembly version: ",
+                    dataView.getUint32(4, true));
+  }
+}
+
+function concatArrayBuffers(...buffers) {
+  const ret = new Uint8Array(buffers.reduce((acc, buff) => acc + buff.byteLength, 0));
+  let currOffset = 0;
+  for (let i = 0; i < buffers.length; i++) {
+    const buffer = buffers[i];
+    ret.set(buffer, currOffset);
+    currOffset += buffer.byteLength;
+  }
+  return ret.buffer;
+}
+
+/**
+ * WebAssembly got that weird "LEB128" integer format.
+ *
+ * This function parses the unsigned version into a JS number.
+ * @param {DataView} dv
+ * @param {number} offset - Offset, in bytes, in the buffer where that number
+ * starts.
+ * @returns {Array.<number>}
+ */
+function readULeb128(dv, offset) {
```

Comment:
That function is tough to follow but u are parsing this?
![image](https://user-images.githubusercontent.com/14330374/116701471-37d21400-a9c8-11eb-8238-0d1a6af5fb97.png)

Reply:
I'm not sure I'm understanding which part of the spec you're referencing, but yes, all integer in a WebAssembly binary file are actually LEB128 (https://en.wikipedia.org/wiki/LEB128) which is variable length, even if a length is indicated like here (u32).